utils/_context/containers.py

```diff
@@ -401,6 +410,7 @@ def __init__(self, host_log_folder, use_proxy=True) -> None:
             },
             ports={self.agent_port: f"{self.agent_port}/tcp"},
             stdout_interface=interfaces.agent_stdout,
+            depends_on=["proxy"],
```

Comment:
Could you make `depends_on` a list of TestedContainer, rather than a list of str? spreading container name in the code will make it harder to maintain.